src/shared/blockdev-util.c

```diff
@@ -167,6 +167,65 @@ static int block_device_get_originating(sd_device *dev, sd_device **ret) {
         return 1; /* found */
 }
 
+int block_device_new_from_fd(int fd, BlockDeviceFlag flags, sd_device **ret) {
+        _cleanup_(sd_device_unrefp) sd_device *dev = NULL;
+        dev_t devnum;
+        int r;
+
+        assert(fd >= 0);
+        assert(ret);
+
+        r = fd_get_devnum(fd, FLAGS_SET(flags, BLOCK_DEVICE_BACKING), &devnum);
```

Comment:
I think it would be nicer to change fd_get_devnum() to just takes BlockDeviceFlag directly, instead of converting to boolean here.